src/configure/configurers/localGithubWorkflowConfigurer.ts

```diff
@@ -9,32 +9,75 @@ import { UserCancelledError } from 'vscode-azureextensionui';
 const Layer = 'LocalGitHubWorkflowConfigurer';
 
 export class LocalGitHubWorkflowConfigurer implements Configurer {
     protected githubClient: GithubClient;
-    private queuedPipelineUrl: string;;
+    private queuedPipelineUrl: string;
+    private controlProvider: ControlProvider;
+    private localGitRepoHelper : LocalGitRepoHelper;
 
-    constructor(azureSession: AzureSession, subscriptionId: string) {
+    constructor(azureSession: AzureSession, subscriptionId: string, localgitRepoHelper: LocalGitRepoHelper) {
+        this.controlProvider = new ControlProvider();
+        this.localGitRepoHelper = localgitRepoHelper;
     }
 
     public async getInputs(inputs: WizardInputs): Promise<void> {
         this.githubClient = new GithubClient(inputs.githubPATToken, inputs.sourceRepository.remoteUrl);
-        return;
+        try {
+            inputs.isNewOrganization = false;
+            if (!inputs.sourceRepository.remoteUrl) {
+                let githubOrganizations = await this.githubClient.listOrganizations(inputs.githubPATToken);
+               
+                if ( githubOrganizations &&  githubOrganizations.length > 0) {
+                    let selectedOrganization = await this.controlProvider.showQuickPick(
+                        constants.SelectGitHubOrganization,
+                        githubOrganizations.map(x => { return { label: x.login }; }),
+                        { placeHolder: Messages.selectGitHubOrganizationName },
+                        TelemetryKeys.OrganizationListCount);
+                        inputs.organizationName = selectedOrganization.label;
+
+
+                    let newGitHubRepo = await this.githubClient.generateGitHubRepository(inputs.organizationName, inputs.sourceRepository.localPath) as unknown as GitHubRepo | void;
+                    if(newGitHubRepo){
+                        inputs.sourceRepository.remoteName = newGitHubRepo.name;
+                        inputs.sourceRepository.remoteUrl = newGitHubRepo.html_url+".git";
+                        inputs.sourceRepository.repositoryId = GitHubProvider.getRepositoryIdFromUrl(inputs.sourceRepository.remoteUrl);
+                        await this.localGitRepoHelper.initializeGitRepository(inputs.sourceRepository.remoteName, inputs.sourceRepository.remoteUrl);
+                        vscode.window.showInformationMessage(utils.format(Messages.newGitHubRepositoryCreated, newGitHubRepo.name));
+                    }
+                    else{
+                        vscode.window.showErrorMessage(Messages.cannotCreateGitHubRepository);
+                        throw Error;
+                    }
+                    
+                }
+                else{
+                    vscode.window.showErrorMessage(Messages.createGitHubOrganization);
+                    throw Error;
+                }
+            }
+        }
+        catch (error) {
+            telemetryHelper.logError(Layer, TracePoints.GetGitHubDetailsFailed, error);
```

Comment:
you have already logged two useful error telemetry above. Do you really want to catch then again? If the chanced of other errors happening are very low, then remove this catch block as it will lead to double error telemetries being logged for same error.

Reply:
Ohh sorry, above we are only showing error and not logging telemetry. It should be fine then.
But do think about this. I think it would be better to log telemetry there only as they are very pointed telemetry point unlike GetGitHubDetailsFailed which is very generic. We can remove this catch block in such a case

Reply:
Yeah, logging the telemetry in the 'else' block will be a better option. And it will also eliminate the need of try-catch block. I'll make the required changes.